server/src/main/java/io/deephaven/server/session/TicketRouter.java

```diff
@@ -309,6 +316,29 @@ public void visitFlightInfo(@Nullable final SessionState session, final Consumer
         byteResolverMap.iterator().forEachRemaining(resolver -> resolver.forAllFlightInfo(session, visitor));
     }
 
+    public void doAction(@Nullable final SessionState session, Action request, Consumer<Result> visitor) {
+        final String type = request.getType();
+        TicketResolver doActionResolver = null;
+        for (TicketResolver resolver : resolvers) {
+            if (!resolver.supportsDoActionType(type)) {
```

Comment:
I think we chatted directly, but there is no general strategy for resolving this in `O(|DoActionResolvers|)` that said, I don't think we care because we only expect that to include FlightSQL in the immediate future. If we somehow get into a state where we have many do action resolvers, we could consider an interface where the resolver tells us more info about itself, which would help in some cases. For example, some resolvers might have a fixed set; others a prefix; and others, not generalizable and would need Predicate<String> (as essentially the generic interface presested here).